### PR TITLE
Add User-Agent header to tmc-auth SDK

### PR DESCRIPTION
**Pull request by cbusby**
# Description

| **Story ID** | **Story Name** |
| ------------ | -------------- |
| #178486439 |  Add User-Agent header to tmc-auth SDK |

***
## Summary of Changes: 
* Sending the User-Agent header to nginx
* Refactored the base exception so that it will be easier to go into common library later.

***
## Review readiness
This pull request is considered a _Work In Progress_ unless all of the following boxes are checked.
Items not required for this PR should be removed from the below list.

- [ ] A secret scan was run:
    - [ ] on every commit
    - [ ] on every push
- [ ] Existing tests of the following types were run:
    - [ ] unit tests
    - [ ] integration tests
- [ ] Application was run to ensure it is still working as expected
- [ ] Examples given to customers still work as expected
- [ ] Code quality gates have been meet:
    - [ ] Code Coverage
    - [ ] Sonar Quality Gate     
- [ ] Ensured no breaking changes to the public interface
- [ ] One of the following has been done:
    - [ ] Documentation was updated
    - [ ] A story was created to update the documentation

***
## PR Reviewer's Checklist
Be sure to give yourself enough time to review the PR.  We are looking for good approvals, not rubber stamped approvals.
- [x] Pull down the branch
- [x] Review the code
- [x] Build the code
- [ ] Run tests, application and/or examples locally
- [x] Review code's coverage
- [ ] Review code's Sonar quality gate

***
## Reviewers for this PR: 
- [ ] @autonomic-tmc/sdk-team
- [ ] @autonomic-tmc/fe-pr-reviewers

### CODE REVIEW

**Review by cbusby — COMMENTED** [review]
🌈

File: src/main/java/com/autonomic/tmc/auth/exception/ProjectProperties.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+@Slf4j
+public class ProjectProperties {
+
+    static ProjectProperties singletonInstance = null;
+
+    Manifest manifest;
+
+    public static final String DEFAULT_NAME = "[ AUTONOMIC ]";
+    public static final String DEFAULT_VERSION = "[ SDK ]";
+    public static final String SDK_NAME = "tmc-auth";
```
Comment (shah-jenish) [inline src/main/java/com/autonomic/tmc/auth/exception/ProjectProperties.java:40]: Thinking about the different ways we are using Manifest values - for exception handling and for user agent

getName()/getVersion() defaults to `[ AUTONOMIC ]/[ SDK ]` for exception handling
getName()/getVersion() defaults to `tmc-auth/unknown` for user agent

I don't recall why we choose defaults for exception handling but would it make sense to change it to use the same value as the user agent?

Comment (matthew-auto) [reply src/main/java/com/autonomic/tmc/auth/exception/ProjectProperties.java:40]: I also don't recall why we chose those defaults. I believe they are easy to parse in the logs. There are a couple of paths in this class now:
 
- `[ AUTONOMIC ] - [ SDK ]` will be printed when there's an issue retrieving the package name and version during an exception path
- `tmc-auth - 1.x.x` will be printed when there is an exception but no issue retrieving the package name and version.
- `tmc-auth/unknown` will be used when there is an issue retrieving the package name and version.

I agree we can refactor and improve this class and am in favor of the simple `tmc-auth/unknown` or ` [ tmc-auth ] - [ unknown ]` - but we can discuss this.

**Review by shah-jenish — APPROVED** [review]

**Review by karthi-au — APPROVED** [review]
Sonar coverage is missing for this PR and for master. Since it's not related to this PR, approving it. We'll talk with the team to find if this is intentional and maybe write a separate story to fix this.

**Pull request merged**